### DISCUSSION TITLE
Opening up funding for ratatui

### DISCUSSION THREAD

**Discussion by mindoodoo** [ratatui/ratatui#206] [Discussion]
Over the past couple months, the need for funding options for this project has been mentioned quite a few times. Off the top of my head, here are a couple things that would require funding in order to implement properly :

- **Ratatui** website hosting (if we were to make one)
- **Matrix Bridge**
- Sponsoring highly active contributors

With that in mind, we'd love to hear everyone's suggestions and discuss some of the popular funding options with everyone here !

Some popular funding options :
- [Github Sponsors](https://docs.github.com/en/sponsors)
- [Libera pay](https://en.liberapay.com/)
- [Open Collective](https://opencollective.com/)
- Grants (?) (Found [this](https://github.com/ralphtheninja/open-funding#grants) list with a quick google)

Finally, we need to setup a clear and structured manner to manage these funds, a couple considerations that come to mind :
- **Achieving full transparency**
- Who
- How (what tools)
- Legal and tax considerations
- Safeguards against mismanagement
- Donor recognition

Maintainers feel free to edit this base message to add considerations / options you think should be taken into account !

**Comment by Siriusmart** [comment]
This [Matrix discord bridge](https://t2bot.io/discord/) is free (but they accept [donations](https://t2bot.io/donations/))

**Reply by mindoodoo** [reply]
Do you have experience with it ?

**Reply by Siriusmart** [reply]
I've bridged some channels from a matrix space to a discord server, it's not complicated.
I can help bridging the chats if you want.

**Comment by sayanarijit** [comment]
Thanks for creating the thread. I believe this is becoming an important step for the project to set up some kind of funding option, specially so that we can offer some reward to the contributors for taking critical responsibilities. I/we don't have much knowledge of how these things work for organisations, so any kind of help/insight from anyone with such experience would be highly appreciated.

**Reply by mindoodoo** [reply]
I like the idea of rewarding contributors with any funds left unused. I would love to see this being a highly dynamic system where a vote can be held every month or so.

**Comment by orhun** [comment]
Thanks for bringing this up! I agree that using funding to support highly active contributors would be a good idea. Also, we can use it for expenses such as `ratatui` website, which I had been planning to set up for a while now. So far I only used GitHub Sponsors and Open Collective from that list and I would say that they are pretty straightforward. The important part is to have _full_ transparency and manage the money somehow. We can look at other projects such as [ruffle](https://github.com/ruffle-rs), [elk](https://github.com/elk-zone) and [neovim](https://github.com/neovim) about how they do it and potentially get some comments from people who are knowledgeable regarding the topic.

**Reply by mindoodoo** [reply]
+1 **full transparency** is an ABSOLUTE must.

**Comment by erlend-sh** [comment]
I highly recommend GH Sponsors are your default funding option. It’s just a natural fit for any project that’s already on GitHub; the networking effects and low fees are too good to pass up.

It can be integrated with OC for oversight: https://docs.opencollective.com/help/collectives/osc-verification/github-sponsors

**Reply by mindoodoo** [reply]
I haven't looked into OC that much yet, but yeah GitHub sponsors seems like a logical first step. If OC has adequate tools for oversight and transparency there's a chance this might be the package we're looking for.

**Comment by mo8it** [comment]
Liberapay would be awesome since Github cancelled paying with Paypal and not everyone has a credit card. Plus Liberapay is free open source software :D

**Reply by TimerErTim** [reply]
Isn't nearly every bank card also a debit card (basically a credit card with imediate expenses instead of collective at end of month) that works just like a credit card?

Or is that only in my country?

**Comment by mdtro** [comment]
+1 to opening up sponsorship!

I don't have much to contribute to the discussion here, but wanted to say I dropped by to find out how to sponsor. I wasn't able to find any immediate path to do so and found this after a quick search.

Of your choices, the most common ones I see used are GitHub Sponsors and Open Collective. I'd love to see this project on either (or both). 😄

**Comment by orhun** [comment]
We are considering to use Open Collective + GitHub Sponsors for funding. We are actively researching this topic - here are some notes from our last meeting: <https://hackmd.io/K_BREUF3QpuEcJOOfHRbjQ>

**Comment by orhun** [comment]
Closing... <https://blog.orhun.dev/open-source-funding-with-ratatui> :)
